hypervisor/arch/x86/guest/ept.c

```diff
@@ -203,56 +203,31 @@ void ept_del_mr(struct acrn_vm *vm, uint64_t *pml4_page, uint64_t gpa, uint64_t
  */
 void ept_flush_leaf_page(uint64_t *pge, uint64_t size)
 {
-	uint64_t flush_base_hpa = INVALID_HPA, flush_end_hpa;
-	void *hva = NULL;
-	uint64_t flush_size = size;
+	uint64_t base_hpa, end_hpa;
 	uint64_t sw_sram_bottom, sw_sram_top;
 
 	if ((*pge & EPT_MT_MASK) != EPT_UNCACHED) {
-		flush_base_hpa = (*pge & (~(size - 1UL)));
-		flush_end_hpa = flush_base_hpa + size;
+		base_hpa = (*pge & (~(size - 1UL)));
+		end_hpa = base_hpa + size;
 
 		 sw_sram_bottom = get_software_sram_base();
 		 sw_sram_top = sw_sram_bottom + get_software_sram_size();
 		/* When Software SRAM is not initialized, both sw_sram_bottom and sw_sram_top is 0,
-		 * so the below if/else will have no use
+		 * so the first if below will have no use.
 		 */
-		if (flush_base_hpa < sw_sram_bottom) {
-			/* Only flush [flush_base_hpa, sw_sram_bottom) and [sw_sram_top, flush_base_hpa),
-			 * ignore [sw_sram_bottom, sw_sram_top)
-			 */
-			if (flush_end_hpa > sw_sram_top) {
-				/* Only flush [flush_base_hpa, sw_sram_bottom) and [sw_sram_top, flush_base_hpa),
-				 * ignore [sw_sram_bottom, sw_sram_top)
-				 */
-				flush_size = sw_sram_bottom - flush_base_hpa;
-				hva = hpa2hva(flush_base_hpa);
-				stac();
-				flush_address_space(hva, flush_size);
-				clac();
-
-				flush_size = flush_end_hpa - sw_sram_top;
-				flush_base_hpa = sw_sram_top;
-			} else if (flush_end_hpa > sw_sram_bottom) {
-				/* Only flush [flush_base_hpa, sw_sram_bottom)
-				 * and ignore [sw_sram_bottom, flush_end_hpa)
-				 */
-				flush_size = sw_sram_bottom - flush_base_hpa;
-			}
-		} else if (flush_base_hpa < sw_sram_top) {
-			if (flush_end_hpa <= sw_sram_top) {
-				flush_size = 0UL;
-			} else {
-				/* Only flush [sw_sram_top, flush_end_hpa) and ignore [flush_base_hpa, sw_sram_top) */
-				flush_base_hpa = sw_sram_top;
-				flush_size = flush_end_hpa - sw_sram_top;
-			}
+		if (base_hpa < sw_sram_bottom) {
+			/* Flush [base_hpa, min(end_hpa, sw_sram_bottom) */
```

Comment:
The original comments are more readable than this.  Please use the style of original comments, and explain the case of situation, i.e. how the regions overlap or not etc...

Reply:
The logic in this patch doesn't  need to care whether the region is overlap with pSRAM or not. What it does is:
if there is one part in the region is  on left side of pSRAM region, flush it first; if there's one part in the region is on right side of pSRAM region, flush it then.